test/cmd/images_tests.sh

```diff
@@ -0,0 +1,245 @@
+repository="$( oc get is/newrepo -o jsonpath='{.status.dockerImageRepository}' )"
+os::cmd::expect_success_and_text 'oc get istag/newrepo:direct -o jsonpath={.image.dockerImageReference}' "^$repository@sha256:"
+os::cmd::expect_success_and_text 'oc get istag/newrepo:direct -o jsonpath={.image.dockerImageReference}' '/test-cmd-images-2/newrepo@sha256:'
+# tagging an image using --reference does not
+os::cmd::expect_success "oc tag $project/test:new newrepo:indirect --reference"
+os::cmd::expect_success_and_text 'oc get istag/newrepo:indirect -o jsonpath={.image.dockerImageReference}' "/$project/test@sha256:"
+os::cmd::expect_success "oc project $project"
+# test scheduled and insecure tagging
+os::cmd::expect_success 'oc tag --source=docker mysql:5.7 newrepo:latest --scheduled'
+os::cmd::expect_success_and_text "oc get is/newrepo --template='{{(index .spec.tags 1).importPolicy.scheduled}}'" 'true'
+os::cmd::expect_success_and_text "oc describe is/newrepo" '\* tag is scheduled'
+os::cmd::expect_success 'oc tag --source=docker mysql:5.7 newrepo:latest --insecure'
+os::cmd::expect_success_and_text "oc describe is/newrepo" '\! tag is insecure'
+os::cmd::expect_success_and_not_text "oc describe is/newrepo" '\* tag is scheduled'
+os::cmd::expect_success_and_text "oc get is/newrepo --template='{{(index .spec.tags 1).importPolicy.insecure}}'" 'true'
+
+# test creating streams that don't exist
+os::cmd::expect_failure_and_text 'oc get imageStreams tagtest3' 'not found'
+os::cmd::expect_failure_and_text 'oc get imageStreams tagtest4' 'not found'
+os::cmd::expect_success 'oc tag mysql:latest tagtest3:latest tagtest4:latest --alias'
+os::cmd::expect_success_and_text "oc get is/tagtest3 --template='{{(index .spec.tags 0).from.kind}}'" 'ImageStreamTag'
+os::cmd::expect_success_and_text "oc get is/tagtest4 --template='{{(index .spec.tags 0).from.kind}}'" 'ImageStreamTag'
+os::cmd::expect_success 'oc delete is/tagtest is/tagtest2 is/tagtest3 is/tagtest4'
+os::cmd::expect_success_and_text 'oc tag mysql:latest tagtest:new1 --alias' 'Tag tagtest:new1 set up to track mysql:latest.'
+os::cmd::expect_success_and_text 'oc tag mysql:latest tagtest:new1' 'Tag tagtest:new1 set to mysql@sha256:'
+
+# test deleting a spec tag using oc tag
+os::cmd::expect_success 'oc create -f test/testdata/test-stream.yaml'
+os::cmd::expect_success_and_text 'oc tag test-stream:latest -d' 'Deleted'
+os::cmd::expect_success 'oc delete is/test-stream'
+echo "tag: ok"
+os::test::junit::declare_suite_end
+
+os::test::junit::declare_suite_start "cmd/images${IMAGES_TESTS_POSTFIX:-}/delete-istag"
+# test deleting a tag using oc delete
+os::cmd::expect_success_and_text "oc get is perl --template '{{(index .spec.tags 0).name}}'" '5.16'
+os::cmd::expect_success_and_text "oc get is perl --template '{{(index .status.tags 0).tag}}'" 'latest'
+os::cmd::expect_success "oc delete istag/perl:5.16 --context='${cluster_admin_context}'"
```

Comment:
@deads2k seeing this fail sometimes .... 
```
FAILURE after 0.086s: test/cmd/images_tests.sh:237: executing 'oc delete istag/perl:5.16 --context='default/127-0-0-1:28443/system:admin'' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
Error from server: imagestreams "perl" not found
```

 I don't understand the non-determinism